lib/services/product-manager.js

```diff
@@ -0,0 +1,102 @@
+import _ from 'lodash'
+import Promise from 'bluebird'
+
+export default class ProductManager {
+  constructor (logger, client) {
+    this.client = client
+    this.logger = logger.child({ service: 'productManager' })
+
+    this.loadBatchCount = 20
+    this.loadConcurrency = 2
+  }
+
+  createProduct (product) {
+    this.logger.debug('Creating product', product)
+
+    return this.client.products
+      .create(product)
+      .then(res => res.body)
+  }
+
+  publishProduct (product) {
+    const actions = [{
+      action: 'publish'
+    }]
+
+    return this.updateProduct(product, actions)
+  }
+
+  updateProduct (product, actions) {
+    const request = {
+      version: product.version,
+      actions
+    }
+
+    return this.client.products
+      .byId(product.id)
+      .update(request)
+  }
+
+  filterOutDuplicateProducts (products) {
+    console.log(products)
+    throw new Error('NOT IMPLEMENTED')
+  }
+
+  async getProductsBySku (skus) {
+    // if skus have more than a certain amount run multiple parallel requests
+    if (skus.length > this.loadBatchCount) {
+      const productBatches = Promise.map(
+        _.chunk(skus, this.loadBatchCount), // load max N skus per one request
+        skuBatch => this.getProductsBySku(skuBatch),
+        { concurrency: this.loadConcurrency } // load products with concurrency
+      )
+
+      return this.filterOutDuplicateProducts(_.flatten(productBatches))
+    }
+
+    const skuPredicate = `${skus.join('","')}`
```

Comment:
Can be `const skuPredicate = skus.join('","')`

Reply:
Updated bf4c30f